plugins/pulp_rpm/plugins/importers/yum/sync.py

```diff
@@ -198,13 +198,15 @@ def get_metadata(self):
         try:
             metadata_files.download_repomd()
         except IOError, e:
-            raise FailedException(str(e))
+            raise PulpCodedException(error_code=error_codes.RPM1004, reason=str(e))
```

Comment:
Ok, I see what's going on here now. It's really a judgement call about which tracebacks could be valuable, and trying to not pollute the log with tracebacks that aren't helpful. The platform will always log the PulpCodedException traceback at debug. In the case below, I think it should be very rare that we fail to parse repo metadata. So additional info about exactly what went wrong might be helpful.

On the other hand, failure to download remote metadata will happen all the time, and we can attach a "reason" message to the error that will very likely be useful. So I didn't think it would be valuable to include that traceback at debug.